src/core/src/CoreMain.py

```diff
@@ -44,11 +44,12 @@ def __init__(self, argv):
 
             # Current operation in status handler is set to either assessment or installation when these operations begin. Setting it to assessment since that is the first operation that runs always.
             # This ensures all errors occurring before assessment starts are logged within the error objects of assessment substatus
-            if status_handler.get_current_operation() is None:
-                status_handler.set_current_operation(Constants.ASSESSMENT)  # Needs to be revisited
+            if status_handler.get_current_operation() is None and not bootstrapper.auto_assessment_only:
+                status_handler.set_current_operation(Constants.ASSESSMENT)
```

Comment:
1. We need to revisit this logic with ConfigurePatching coming in -- Configure patching does not execute under auto-assessment only mode. For the normal case, this can be addressed later - this PR doesn't change any behavior here.

2. Auto assessment errors are tracked in the Assessment substatus when it start it. Anything before that goes into logs and telemetry. This is by design given how auto assessment works.